crates/sui-types/src/message_envelope.rs

```diff
@@ -148,16 +149,23 @@ impl<T: Message> Envelope<T, EmptySignInfo> {
         }
     }
 
-    pub fn verify_signature(&self) -> SuiResult {
-        self.data.verify(None)
+    pub fn verify_signature(&self, verify_params: &VerifyParams) -> SuiResult {
+        self.data.verify(verify_params)
     }
 
-    pub fn verify(self) -> SuiResult<VerifiedEnvelope<T, EmptySignInfo>> {
-        self.verify_signature()?;
+    pub fn verify_with_params(
```

Comment:
Not your fault here, but please add a TODO to rename these two functions as followup?